app/views/components/gds/PaginationUtil.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.components.gds
+
+import java.lang.Math.abs
+
+import views.components.gds.PaginationUtil.PaginationItem._
+
+object PaginationUtil {
+
+  def paginationElements(currentPageIndex: Int, pagesTotal: Int, neighbourPagesAmount: Int = 1): Seq[PaginationItem] = {
```

Comment:
Is not beter to take here Paginated ? Most of those checks would be gone in this case - parameters are passed exactly from paginated

Reply:
Not `Paginated` but `Page` and it would only reduce the first two parameters.
It is sensible to do, but I'd probably then need to implement all necessary checks in `Page` class as it has none at the moment.